src/iris/ui/static/js/iris.js

```diff
@@ -2321,15 +2331,22 @@ iris = {
                 contentType: 'text'
             }).done(function(){
                 iris.createAlert('Modal settings saved succesfully', 'success');
-                if (self.data.settings.customApp.indexOf(app) === -1) {
-                    self.data.settings.customApp.push(app);
+                if (!self.data.settings.customApp.has(app)) {
```

Comment:
the `! .has(app)` check before `.add(app)` isn't needed, as sets silently ignore dupes